### PR TITLE
Updated Slow Tests and Tuning Scripts, Cleaned Some pytorch_dataset Documentation, Improved EIC Quantile Binning

### PR DESCRIPTION
**Pull request by Oufattole**
This should be merged after #67 

This pull request includes the following updates:

1. Cleaned up docstrings for the pytorch_dataset class
2. Integrated necessary configuration changes to support tuning
3. Improved Quantile Binning for EIC:
   - Updated quantile binning for EIC to exclude non-numeric bins for vocab indices.
4. Refactored `tune.py` Job Launching Script:
   - Made `tune.py` more DRY by consolidating training and finetuning logic from `train.py` and `finetune.py`.
   - Resolved a bug with checkpointing in hyperparameter tuning:
     - Disabled PyTorch Lightning's model checkpoint callback.
     - Used Ray Tune’s checkpointing to avoid conflicts with multiple trials writing to the same directory.
5. Updated slow finetuning and CLI tests to handle distributed finetuning and support multiseed and hyperparameter sweeps effectively. These are not run by github actions as it skips these slow guys.

### CODE REVIEW

**Comment by mmcdermott** [comment]
@Oufattole if this should be merged after #67, then do you just want to merge this into branch 55 instead of dev?

File: src/meds_torch/data/datamodule.py
Comment (mmcdermott) [inline src/meds_torch/data/datamodule.py:112]: why use different names for stages vs. splits?

File: src/meds_torch/data/datamodule.py
Comment (mmcdermott) [inline src/meds_torch/data/datamodule.py:158]: Why do you need this method if it is empty?

File: src/meds_torch/tune.py
Comment (mmcdermott) [inline src/meds_torch/tune.py:132]: This doesn't seem like something you should globally overwrite.

File: src/meds_torch/tune.py
Comment (mmcdermott) [inline src/meds_torch/tune.py:134]: I would recommend setting `cfg.paths.time_output_dir` to a `Path` once with a variable, then using the `pathlib` approach to make the directory.

**Review by mmcdermott — APPROVED** [review]

**Comment by Oufattole** [comment]
> @Oufattole if this should be merged after #67, then do you just want to merge this into branch 55 instead of dev?

I have a sequence of autoregressive modeling pull requests that I will convert from drafts that rely on these changes, so I am hesitant to not merge it dev. What do you think, maybe we can do a double merge (-> 55 -> dev)?

File: src/meds_torch/tune.py
Comment (Oufattole) [reply src/meds_torch/tune.py:132]: Oh [that disables the RAY memory monitor](https://docs.ray.io/en/latest/ray-core/scheduling/ray-oom-prevention.html) which often kills my jobs.

Comment (Oufattole) [reply src/meds_torch/tune.py:132]: I think the best solution is to add an arg:
`os.environ["RAY_memory_monitor_refresh_ms"] = cfg.ray_memory_monitor_refresh_ms`

File: src/meds_torch/tune.py
Comment (Oufattole) [reply src/meds_torch/tune.py:134]: I was considering in hydra switching:
```yaml
time_output_dir: ${paths.output_dir}/${now:%Y-%m-%d_%H-%M-%S_%f}
```

to 
```yaml
time_output_dir:
    _target_: pathlib.Path
    _args_:
        - ${paths.output_dir}/${now:%Y-%m-%d_%H-%M-%S_%f}
```

Thoughts @mmcdermott ?

Comment (mmcdermott) [reply src/meds_torch/tune.py:134]: It will make your config file code harder to read but it isn't crazy. I think either is fine.

File: src/meds_torch/data/datamodule.py
Comment (Oufattole) [reply src/meds_torch/data/datamodule.py:112]: The reason for this different naming scheme is to bridge the different naming schemes used in pytorch lightning vs meds-transform. But depending on how a user preprocessed data, any split name may be used.

I propose in the [data schema ](https://github.com/Oufattole/meds-torch/blob/a35a3a3917b0c7be7db7dc35c9a06010edbb45a3/src/meds_torch/configs/data/pytorch_dataset.yaml#L36) we have a split map added so users can define the proper splits they want:

```yaml
split_names
    train: train
    validate: tuning
    test: held_out
```

File: src/meds_torch/tune.py
Comment (Oufattole) [reply src/meds_torch/tune.py:134]: This would complicate overriding as well, so it would be better to just use pathlib as needed in the codebase

Comment (Oufattole) [reply src/meds_torch/tune.py]: I added the `time_output_dir` conversion from str -> Path in the task_wrapper decorator, which applies it to all scripts

**Pull request merged**
